app/forms/claims/claim_window_form.rb

```diff
@@ -0,0 +1,59 @@
+class Claims::ClaimWindowForm < ApplicationForm
+  attr_accessor :id
+
+  date_attribute :starts_on
+  date_attribute :ends_on
+
+  attribute :academic_year_id
+
+  validates :starts_on, presence: true
+  validates :ends_on, presence: true, comparison: { greater_than_or_equal_to: :starts_on }
+  validates :academic_year_id, presence: true
+
+  validate :does_not_start_within_another_claim_window
+  validate :does_not_end_within_another_claim_window
+
+  def initialize(attributes = {})
+    super
+
+    if id
+      self.starts_on ||= claim_window.starts_on
+      self.ends_on ||= claim_window.ends_on
+      self.academic_year_id ||= claim_window.academic_year_id
+    end
+  end
+
+  def persist
+    claim_window.update!(starts_on:, ends_on:, academic_year_id:)
+  end
+
+  def new_back_path
+    new_claims_support_claim_window_path(claims_claim_window_form: slice(:starts_on, :ends_on, :academic_year_id))
+  end
+
+  def edit_back_path
+    edit_claims_support_claim_window_path(id, claims_claim_window_form: slice(:starts_on, :ends_on, :academic_year_id))
+  end
```

Comment:
Knowledge of routing and path helpers feels like a responsibility for the Controller layer.

For example, this is how we've handled it in Placements:

https://github.com/DFE-Digital/itt-mentor-services/blob/c7a56194a1cc3e249e967a2456aabb407b5afba9/app/controllers/placements/schools/mentors/add_mentor_controller.rb#L37-L55

However I appreciate Form Objects are intrinsically coupled to the View layer, so these methods aren't necessarily introducing additional coupling.

Not a strong opinion – no change necessarily required.

Reply:
We'll leave this as is for now and revisit in the future if needed.